pkg/trace/local_tracer.go

```diff
@@ -89,18 +90,45 @@ func NewLocalTracer(cfg *config.Config, logger log.Logger, chainID, nodeID strin
 	if cfg.Instrumentation.TracePullAddress != "" {
 		go lt.servePullData()
 	}
+
 	if cfg.Instrumentation.TracePushConfig != "" {
 		s3Config, err := readS3Config(path.Join(cfg.RootDir, "config", cfg.Instrumentation.TracePushConfig))
 		if err != nil {
 			return nil, fmt.Errorf("failed to read s3 config: %w", err)
 		}
 		lt.s3Config = s3Config
 		go lt.pushLoop()
+	} else if s3Config, err := GetPushConfigFromEnv(); err == nil {
```

Comment:
In the case of setting 4 environment variables, but forgetting the fifth, this will consider the env variables not set at all. Do we want this? Or we want to account for the case where if any of the env variables is set, we return an error telling the user to pass the rest correctly or remove the one that's set?

I don't know how this will be used, if it's just a temporary functionality for testing or will long live in the repo, so  won't block on this.

Reply:
Yes, all the fields should be present to start pushing traced data to the remote server. 

As for providing feedback on the missing env variables, I can create an issue to discuss it further.